internal/service/vpc/resource_aws_vpc_peering_connection.go

```diff
@@ -0,0 +1,337 @@
+				stateInfoToString(pc.StateInfo))...)
+		default:
+			return diag.Errorf("Unknown VPC peering connection cannot state: %s", pc.State)
+		}
+	}
+
+	d.SetId(schemautil.BuildResourceID(projectName, vpcID, pc.PeerCloudAccount, pc.PeerVPC, *pc.PeerRegion))
+
+	// in case of an error delete VPC peering connection
+	if diags.HasError() {
+		return append(diags, resourceAWSVPCPeeringConnectionDelete(ctx, d, m)...)
+	}
+
+	return append(diags, resourceAWSVPCPeeringConnectionRead(ctx, d, m)...)
+}
+
+func resourceAWSVPCPeeringConnectionRead(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*aiven.Client)
+
+	projectName, vpcID, peerCloudAccount, peerVPC, peerRegion := parsePeeringVPCId(d.Id())
+	pc, err := client.VPCPeeringConnections.GetVPCPeering(
+		projectName, vpcID, peerCloudAccount, peerVPC, peerRegion)
+	if err != nil {
+		return diag.FromErr(schemautil.ResourceReadHandleNotFound(err, d))
+	}
+
+	return copyAWSVPCPeeringConnectionPropertiesFromAPIResponseToTerraform(d, pc, projectName, vpcID)
+}
+
+func resourceAWSVPCPeeringConnectionDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*aiven.Client)
+
+	projectName, vpcID, peerCloudAccount, peerVPC, peerRegion := parsePeeringVPCId(d.Id())
+
+	err := client.VPCPeeringConnections.DeleteVPCPeering(
+		projectName,
+		vpcID,
+		peerCloudAccount,
+		peerVPC,
+		peerRegion,
+	)
+	if err != nil && !aiven.IsNotFound(err) {
+		return diag.Errorf("Error deleting VPC peering connection: %s", err)
+	}
+
+	stateChangeConf := &resource.StateChangeConf{
+		Pending: []string{
+			"ACTIVE",
+			"APPROVED",
+			"APPROVED_PEER_REQUESTED",
+			"DELETING",
+			"INVALID_SPECIFICATION",
+			"PENDING_PEER",
+			"REJECTED_BY_PEER",
+			"DELETED_BY_PEER",
+		},
+		Target: []string{
+			"DELETED",
+		},
+		Refresh: func() (interface{}, string, error) {
+			pc, err := client.VPCPeeringConnections.GetVPCPeering(
+				projectName,
+				vpcID,
+				peerCloudAccount,
+				peerVPC,
+				peerRegion,
+			)
+			if err != nil {
+				return nil, "", err
+			}
+			return pc, pc.State, nil
+		},
+		Delay:      10 * time.Second,
+		Timeout:    d.Timeout(schema.TimeoutDelete),
+		MinTimeout: 2 * time.Second,
+	}
+	if _, err := stateChangeConf.WaitForStateContext(ctx); err != nil && !aiven.IsNotFound(err) {
+		return diag.Errorf("Error waiting for Aiven AWS VPC Peering Connection to be DELETED: %s", err)
+	}
+	return nil
+}
+
+func resourceAWSVPCPeeringConnectionImport(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	if len(strings.Split(d.Id(), "/")) != 3 {
+		return nil, fmt.Errorf("invalid identifier %v, expected <project_name>/<vpc_id>/<aws_account_id>", d.Id())
+	}
+
+	dig := resourceAWSVPCPeeringConnectionRead(ctx, d, m)
+	if dig.HasError() {
+		return nil, errors.New("cannot get aws vpc peering connection")
```

Comment:
Probably better to use `AWS VPC`

Reply:
I agree, though Go convention is to use only lowercase letters when constructing errors